### DISCUSSION TITLE
Disable argocd from automatically adding labels to configmap.

### DISCUSSION THREAD

**Discussion by ghost** [argoproj/argo-cd#19010] [Q&A]
I use Kustomization build
When Kustomization builds, the common base configmap will be automatically added to labels, which will cause another app outofsync

[k8s yaml](https://gitlab.com/doubao2/doubao.git)

Two different deploy will reference the same configmap.

I created two apps in argocd. 
When I deploy one of the apps synchronously, the other app gets outofsync.

![image](https://github.com/argoproj/argo-cd/assets/55606580/7539ac07-949c-4603-8efc-b14fd5f762ba)

![image](https://github.com/argoproj/argo-cd/assets/55606580/230667fd-fc1f-4571-a775-e435b8479097)

The reason for outofsync is that argocd automatically adds labels.

I've searched a lot, github, argocd website but haven't found out how to disable.

Finally found a way to ignore labels.

But this one still has the SharedResourceWarning

~~~yaml
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
  name: argocd-cm
  namespace: argocd
data:
  resource.customizations.ignoreDifferences.all: |
    managedFieldsManagers:
    - kube-controller-manager
    jsonPointers:
    - /metadata/labels/app.kubernetes.io~1instance
~~~

**Comment by The32Beginner** [comment]
Hi, 
I was struggling with the same issue.
Trying to use just one configmap file for several deployments.
and I solved it following your advice. Thanks.

I also found out, we could just set the options on ArgoCD App.
(https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/)

I just added under 'spec' parts like below.
```
  ignoreDifferences:
  - group: '*'
    jsonPointers:
    - /metadata/labels
    kind: ConfigMap
```
and boom! 
(However, warning is still appears on the top right 😂)
Thank you again.
